### DISCUSSION TITLE
should i lear dsa ?

### DISCUSSION THREAD

**Discussion by newamyw1** [community/community#68120] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

should i have to learn data structure and algo before math

**Comment by emranpy** [comment]
sure you gotta learn math discrete math to be able to think like a programmer

**Comment by sudhanshu-77** [comment]
you should have basic mathematics  knowledge before start dsa and algorithms it will help you to learn concepts easily.

**Comment by Akash1134** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
